src/uuid/rfc4122.cr

```diff
@@ -0,0 +1,86 @@
+# Support for RFC 4122 UUID variant.
+struct UUID
+  # RFC 4122 UUID variant versions.
+  enum Version
+    # Unknown version.
+    Unknown = 0
+
+    # Version 1 - date-time and MAC address.
+    V1 = 1
+
+    # Version 2 - DCE security.
+    V2 = 2
+
+    # Version 3 - MD5 hash and namespace.
+    V3 = 3
+
+    # Version 4 - random.
+    V4 = 4
+
+    # Version 5 - SHA1 hash and namespace.
+    V5 = 5
+  end
+
+  # Generates RFC 4122 UUID `variant` with specified `version`.
+#  def initialize(@bytes : StaticArray(UInt8, 16), version = Version::V4)
+#    #@bytes.to_unsafe.copy_from SecureRandom.random_bytes(16).pointer(16), 16
+#
+#    case version
+#    when Version::V4
+#      variant = Variant::RFC4122
+#      version = Version::V4
+#    else
+#      raise ArgumentError.new "Creating #{version} not supported."
+#    end
+#  end
+
+  # Returns version based on provided 6th `byte` (0-indexed).
+  def self.byte_version(byte : UInt8)
```

Comment:
The `.byte_version(byte)` is only ever called from `#version`. The `case` should be moved to `#version` instead.